collate/collate.go

```diff
@@ -102,90 +102,28 @@ func (b *Buffer) Reset() {
 
 // Compare returns an integer comparing the two byte slices.
 // The result will be 0 if a==b, -1 if a < b, and +1 if a > b.
+// Note that this is less performant than calling c.Sort() because
+// a new buffer will be allocated for each call.
 func (c *Collator) Compare(a, b []byte) int {
-	// TODO: skip identical prefixes once we have a fast way to detect if a rune is
-	// part of a contraction. This would lead to roughly a 10% speedup for the colcmp regtest.
-	c.iter(0).SetInput(a)
-	c.iter(1).SetInput(b)
-	if res := c.compare(); res != 0 {
-		return res
-	}
-	if !c.ignore[colltab.Identity] {
-		return bytes.Compare(a, b)
-	}
-	return 0
+	var (
+		buf Buffer
```

Comment:
We might want to get rid of the use of Buffer, to avoid the 4000 byte allocation on each call?  AFAICT that 4000 byte array isn't even used...or maybe is just as a strange way to pre-allocate the capacity of the `keys` slice (vs `make([]byte, 0, 4000)`).  Or maybe we can use one that's a part of the Collator struct, and just call Reset() on it each time?

Reply:
fixed by using a shared buf on the Collator